giza/agents/integrations/uniswap/nft_manager.py

```diff
@@ -0,0 +1,173 @@
+import os
+import time
+
+from ape import Contract
+
+from giza.agents.integrations.uniswap.constants import MAX_UINT_128
+from giza.agents.integrations.uniswap.utils import (
+    calc_amount0,
+    calc_amount1,
+    liquidity0,
+    liquidity1,
+    nearest_tick,
+    price_to_sqrtp,
+    price_to_tick,
+)
+
+
+class NFTManager:
+    def __init__(self, address: str, sender: str):
+        self.contract = Contract(
+            address,
+            abi=os.path.join(os.path.dirname(__file__), "assets/nft_manager.json"),
+        )
+        self.sender = sender
+
+    def get_all_user_positions(self, user_address: str = None):
+        if user_address is None:
+            user_address = self.sender
+        n_positions = self.contract.balanceOf(user_address)
+        positions = []
+        for n in range(n_positions):
+            position = self.contract.tokenOfOwnerByIndex(user_address, n)
+            positions.append(position)
+        return positions
+
+    def close_position(self, nft_id: int, user_address: str = None):
+        if user_address is None:
+            user_address = self.sender
+        liquidity = self.contract.positions(nft_id)["liquidity"]
+        if liquidity > 0:
+            self.decrease_liquidity(nft_id, liquidity=liquidity)
+            self.collect_fees(nft_id, user_address=user_address)
+            self.contract.burn(nft_id, sender=self.sender)
+
+    def collect_fees(
+        self,
+        nft_id: int,
+        user_address: str = None,
+        amount0_max: int = MAX_UINT_128,
+        amount1_max: int = MAX_UINT_128,
+    ):
+        if user_address is None:
+            user_address = self.sender
+        receipt = self.contract.collect(
+            (nft_id, user_address, amount0_max, amount1_max), sender=self.sender
+        )
+        return receipt
+
+    def decrease_liquidity(
+        self,
+        nft_id: int,
+        liquidity: int = None,
+        amount0Min: int = 0,
+        amount1Min: int = 0,
+        deadline: int = None,
+    ):
+        if liquidity is None:
+            liquidity = self.get_pos_info(nft_id)["liquidity"]
+        if deadline is None:
+            deadline = int(time.time() + 60)
+        receipt = self.contract.decreaseLiquidity(
+            (nft_id, liquidity, amount0Min, amount1Min, deadline), sender=self.sender
+        )
+        return receipt
+
+    def add_liquidity(
+        self,
+        nft_id: int,
+        amount0Desired: int,
+        amount1Desired: int,
+        amount0Min: int = 0,
+        amount1Min: int = 0,
+        deadline: int = None,
+    ):
+        pos = self.contract.positions(nft_id)
+        token0 = Contract(
+            pos["token0"],
+            abi=os.path.join(os.path.dirname(__file__), "assets/erc20.json"),
+        )
+        token1 = Contract(
+            pos["token1"],
+            abi=os.path.join(os.path.dirname(__file__), "assets/erc20.json"),
```

Comment:
You will need to include the json files in the wheel: https://python-poetry.org/docs/pyproject/#include-and-exclude

A good way to test this is to create the wheel and check the contents:

```
poetry build

unzip -l dist/{package name}.whl
```

Example:

```
❯ poetry build
Building giza-cli (0.17.0)
  - Building sdist
  - Built giza_cli-0.17.0.tar.gz
  - Building wheel
  - Built giza_cli-0.17.0-py3-none-any.whl
❯ unzip -l dist/giza_cli-0.17.0-py3-none-any.whl
Archive:  dist/giza_cli-0.17.0-py3-none-any.whl
  Length      Date    Time    Name
---------  ---------- -----   ----
      126  01-01-1980 00:00   giza/cli/__init__.py
       81  01-01-1980 00:00   giza/cli/__main__.py
     1047  01-01-1980 00:00   giza/cli/callbacks.py
     7592  01-01-1980 00:00   giza/cli/cli.py
    52627  01-01-1980 00:00   giza/cli/client.py
```